kore/src/Kore/Step/Simplification/Overloading.hs

```diff
@@ -144,13 +167,47 @@ unifyOverloading termPair = case termPair of
         (Inj_ inj@Inj { injChild = App_ firstHead firstChildren })
         (Inj_ inj'@Inj { injChild = App_ secondHead secondChildren })
       | injFrom inj /= injFrom inj' -- this case should have been handled by now
-        -> unifyOverloadingCommonOverload
+        -> Simple <$> unifyOverloadingCommonOverload
             (Application firstHead firstChildren)
             (Application secondHead secondChildren)
             inj { injChild = () }
-    (Pair (App_ firstHead _) (Inj_ Inj { injChild })) ->
+    Pair
+        firstTerm@(App_ firstHead _)
+        (Inj_ inj@Inj { injChild = ElemVar_ secondVar})
+        -> unifyOverloadingVsOverloadedVariable
+            firstHead
+            firstTerm
+            secondVar
+            (Attribute.freeVariables firstTerm)
+            inj { injChild = () }
+    Pair -- it's ok to interchange them here, as this becomes error for matching
```

Comment:
Instead of defining the flipped matches here, let's have a `worker` function that matches in one direction only, and use `catchE` to catch `NotApplicable` and retry with the flipped pair.